### PR TITLE
Bug fixes, refactoring, code cleaning, comments, pylint, and minor changes to GUI

### PR DESCRIPTION
**Pull request by xinyutan17**
New Features
- creating a new experiment via the '-n' flag now copies hyperparams (and targets, if they exist) from the previous experiment directory to the new experiment directory
- test policy samples using 'gps python/gps/gps_main.py --policy N' (takes N samples using the most recent algorithm state), BADMM only
- added documentation for creating an alias 'gps' in place of 'gps python/gps/gps_main.py'
- added documentation for gui configuration options
- CostSum now displays what it is a sum of, in the algorithm information displayed in the gui
- created a gui_demos.py file in python/tests/tests_gui to demonstrate how to use gui components
- new gui configuration options: 'display images' controls whether or not to show image visualizer, 'figsize' controls the default figure size, 'target_output_fontsize' and 'algthm_output_fontsize' control the text fontsize, and 'algthm_output_max_display_size' controls the maximum number of lines displayed in the algorithm output textbox (useful since matplotlib doesn't handle word wrapping)
- mannequin mode in Target Setup now works
- "Calculating." now transitions from "Calculating." to "Calculating.." to "Calculating..."

Refactoring / Bug Fixes
- fixed tfcontroller to work with ISO C++
- fixed bug with moving to initial and move to target, in Target Setup
- fixed typos in documentation
- renamed ActionAxis to ActionPanel, OutputAxis to Textbox, and RealTimePlotter to RealtimePlotter
- significant refactoring of ActionPanel to remove repetitive code, and GPSTrainingGUI to split the update function into a few separate functions
- cleaned up code, added comments, and pylinted files

### CODE REVIEW

File: docs/usage.md [hunk 1]
```diff
@@ -47,3 +47,13 @@ For your reference, your experiments folder contains the following:
   * `log.txt` - the log text of output from the Target Setup GUI and the GPS Training GUI.
   * `targets.npz` - the initial and target state used for this experiment (ROS agent only -- set for other agents in hyperparams.py)
 
+To shorten gps commands, we suggest that you create an alias in your bashrc:
+```sh
+echo "alias gps='python python/gps/gps_main.py'" >> ~/.bashrc
+cd /path/to/gps
+gps -h                      # help
+gps <EXPERIMENT_NAME> -n    # new experiment
+gps <EXPERIMENT_NAME> -     # target setup
```
Comment (cbfinn) [inline docs/usage.md]: missing a -t for target setup.

File: docs/usage.md [hunk 2]
```diff
@@ -47,3 +47,13 @@ For your reference, your experiments folder contains the following:
...
+gps <EXPERIMENT_NAME>       # run training
+gps <EXPERIMENT_NAME> -r N  # resume training
+```
```
Comment (cbfinn) [inline docs/usage.md:65]: Can you add one more option to gps_main that runs a learned policy? Something like the command "gps <EXPERIMENT_NAME> --test N" (which takes N policy samples, assuming that the policy exists)

File: python/gps/gui/config.py [hunk 3]
```diff
@@ -86,59 +39,92 @@
...
+    'initial_mode': 'run',
+    'algthm_output_fontsize': 10,
+    'algthm_output_max_display_size': 15,
 }
 
-gps_training = {
-    'gps_training_log_filename': 'gps_training_log.txt',
-    'image_actuator': target_setup['actuator_names'][0],    # which actuator to get initial and target images from
-}
+def generate_experiment_info(config):
```
Comment (cbfinn) [inline python/gps/gui/config.py:111]: nice to have this code not in the hyperparams - thanks!

**Comment by cbfinn** [comment]
looks great Dennis! only a couple minor comments. let me know if you want me to look at it again.

File: docs/usage.md (hunk 2 above)
Comment (xinyutan17) [inline docs/usage.md:65]: As in, use the most recent version of the algorithm object and take N policy samples with it?

**Comment by xinyutan17** [comment]
I'm waiting to test it in lab on Monday before merging

File: docs/usage.md (hunk 2 above)
Comment (cbfinn) [inline docs/usage.md:65]: Yes -- use the latest iteration from the experiment <experiment name> and take N policy samples, with N defaulting to 1, if it is not specified, and printing out an error if there is no policy (e.g. not running BADMM algorithm)

**Pull request merged**